classes/SpecificPrice.php

```diff
@@ -227,20 +227,34 @@ protected static function filterOutField($field_name, $field_value, $threshold =
         $key_cache     = __FUNCTION__.'-'.$field_name.'-'.$threshold;
         $specific_list = array();
         if (!array_key_exists($key_cache, SpecificPrice::$_filterOutCache)) {
-            $query_count    = 'SELECT COUNT(DISTINCT `'.$name.'`) FROM `'._DB_PREFIX_.'specific_price` WHERE `'.$name.'` != 0';
+            //First we need a base sql sentence
+            $base_sql = ' FROM `'._DB_PREFIX_.'specific_price` WHERE `'.$name.'` != 0';
+            if($check_dates) {
+                $base_sql .= ' AND ( (`from` = \'0000-00-00 00:00:00\' OR `from` <= CURDATE() ) AND (`to` = \'0000-00-00 00:00:00\' OR `to` >= CURDATE() ))';
+            }
+                
+            $query_count    = 'SELECT COUNT(DISTINCT `'.$name.'`)'.$base_sql;
             $specific_count = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($query_count);
             if ($specific_count == 0) {
                 self::$_no_specific_values[$field_name] = true;
 
                 return $query_extra;
             }
-            if ($specific_count < $threshold) {
-                $query             = 'SELECT DISTINCT `'.$name.'` FROM `'._DB_PREFIX_.'specific_price` WHERE `'.$name.'` != 0';
+            
+            //Checking bad values
+            $threshold = (int)$threshold;
```

Comment:
I suggest you sanitize `$threshold` at the very top of the method instead of here (like `$name` is escaped at the very top)